datafusion/physical-plan/src/coalesce_batches.rs

```diff
@@ -346,7 +316,7 @@ impl CoalesceBatchesStream {
                 }
                 CoalesceBatchesStreamState::Exhausted => {
                     // Handle the end of the input stream.
-                    return if self.coalescer.buffer.is_empty() {
+                    return if self.coalescer.is_empty() {
```

Comment:
Putting the BatchCoalescer into its own module means this code must use an accessor rather than directly look at the coalescer's state, which I think improves the modularity (a tiny bit)